test/parallel/test-tls-connect-given-socket.js

```diff
@@ -72,14 +66,14 @@ const server = tls.createServer(options, (socket) => {
 
   // Immediate death socket
   const immediateDeath = net.connect(port);
-  establish(immediateDeath).destroy();
+  establish(immediateDeath, 0).destroy();
```

Comment:
`undefined` will result in the default of `1`. I think either we need to be OK with sending `0` to `common.mustCall()` (I am, at least in this case) or else add some logic to switch out whether `common.mustCall()` or `common.mustNotCall()` gets called. That works too, although might be a bit much. I'm fine with either.